ocaml/typing/env.ml

```diff
@@ -3889,8 +3889,8 @@ let report_lookup_error _loc env ppf = function
   | Unbound_cltype lid ->
       fprintf ppf "Unbound class type %a" !print_longident lid;
       begin match lid with
-      | Lident "float" ->
-        Misc.did_you_mean ppf (fun () -> ["float#"])
+      | Lident name when List.exists (String.equal (name^"#")) (extract_types None env) ->
```

Comment:
Not a new bug in this patch, but this extra check is a regression from existing behavior. With upstream OCaml, if I say

```
class type floot = object end
class type c = float
```

I'm helpfully told I might have meant `floot`. With our HEAD, I'm asked if I meant `float#`, which actually makes no sense in a class type.

Better would be to remove this code (probably by reverting the earlier patch, which was done in haste) and instead add some logic to the `Ptyp_class` case in Typetexp. That proceeds by first trying to find a type that is a variant. If it's not a variant, then we look up `#t`. If that fails, too, we should see if `t#` is in scope and suggest it. If not, then fail with the existing `Env.lookup_cltype` behavior.

Reply:
As suggested, I reverted the old patch and moved the unboxed type checking logic into the `Ptyp_class` case inside Typetexp. The regression is also added as an extra test.

https://github.com/ocaml-flambda/flambda-backend/blob/fc4d1be345b958cb289b5b02976dc714f15d9065/ocaml/testsuite/tests/typing-layouts-float64/parsing.ml#L254-L274

Compare to before, we lose the `Unbound class type float` line in the error message. Should that information be preserved?